### PR TITLE
[Bugfix] Fix logit soft cap in flash-attn backend

### LINKED ISSUE #7419: [Bug]: `gemma2` does not work with flash attention
### 🐛 Describe the bug

Running latest nightly, I see different results for FlashInfer vs FlashAttention Backend even after logits soft capping landing

Install:
```bash
export VLLM_VERSION=0.5.4
pip install [https://vllm-wheels.s3.us-west-2.amazonaws.com/nightly/vllm-${VLLM_VERSION}-cp38-abi3-manylinux1_x86_64.whl](https://vllm-wheels.s3.us-west-2.amazonaws.com/nightly/vllm-$%7BVLLM_VERSION%7D-cp38-abi3-manylinux1_x86_64.whl)
pip install lm_eval==0.4.3
pip install https://github.com/flashinfer-ai/flashinfer/releases/download/v0.1.2/flashinfer-0.1.2+cu121torch2.4-cp310-cp310-linux_x86_64.whl
```

- flash attention
```bash
MODEL=google/gemma-2-27b-it
lm_eval --model vllm --model_args pretrained=$MODEL,add_bos_token=true --tasks gsm8k --num_fewshot 5 --limit 250 --batch_size "auto"


|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.196|±  |0.0252|
|     |       |strict-match    |     5|exact_match|↑  |0.192|±  |0.0250|
```

- flash infer
```bash
MODEL=google/gemma-2-27b-it
VLLM_ATTENTION_BACKEND=FLASHINFER lm_eval --model vllm --model_args pretrained=$MODEL,add_bos_token=true --tasks gsm8k --num_fewshot 5 --limit 250 --batch_size "auto"

vllm (pretrained=google/gemma-2-27b-it,add_bos_token=true), gen_kwargs: (None), limit: 250.0, num_fewshot: 5, batch_size: auto
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.864|±  |0.0217|
|     |       |strict-match    |     5|exact_match|↑  |0.848|±  |0.0228|
```

cc @WoosukKwon

### PR DESCRIPTION
**Pull request by WoosukKwon**
Fixes #7419

After this PR, the GSM8K result looks correct:

```
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.868|±  |0.0215|
|     |       |strict-match    |     5|exact_match|↑  |0.860|±  |0.0220|
```

### CODE REVIEW

**Review by robertgshaw2-redhat — APPROVED** [review]
LGTM! Thanks
